src/EnergyPlus/ExtendedHI.cc

```diff
@@ -0,0 +1,538 @@
+    constexpr Real64 phi_salt = 0.9;                  //           , vapor saturation pressure level of saline solution, steadman1979
+    constexpr Real64 Tc = 310.;                       // K         , core temperature, steadman1979
+    Real64 Pc = phi_salt * pvstar(Tc);                //           , core vapor pressure
+    Real64 L = Le(310.);                              //           , latent heat of vaporization at 310 K
+    constexpr Real64 p = 1.013e5;                     // Pa        , atmospheric pressure
+    constexpr Real64 eta = 1.43e-6;                   // kg/J      , "inhaled mass" / "metabolic rate", steadman1979
+    constexpr Real64 Pa0 = 1.6e3;                     // Pa        , reference air vapor pressure in regions III, IV, V, VI, steadman1979
+
+    // Function to calculate respiratory heat loss, W/m^2
+    Real64 Qv(Real64 Ta, Real64 Pa)
+    {
+        return eta * Q * (cpa * (Tc - Ta) + L * rgasa / (p * rgasv) * (Pc - Pa));
+    }
+
+    // Function to calculate mass transfer resistance through skin, Pa m^2/W
+    Real64 Zs(Real64 Rs)
+    {
+        return (Rs == 0.0387) ? 52.1 : 6.0e8 * std::pow(Rs, 5);
+    }
+
+    // Function to calculate heat transfer resistance through air, exposed part of skin, K m^2/W
+    Real64 Ra(Real64 Ts, Real64 Ta)
+    {
+        constexpr Real64 hc = 17.4;
+        constexpr Real64 phi_rad = 0.85;
+        Real64 hr = epsilon * phi_rad * sigma * (std::pow(Ts, 2) + std::pow(Ta, 2)) * (Ts + Ta);
+        return 1.0 / (hc + hr);
+    }
+
+    // Function to calculate heat transfer resistance through air, clothed part of skin, K m^2/W
+    Real64 Ra_bar(Real64 Tf, Real64 Ta)
+    {
+        constexpr Real64 hc = 11.6;
+        constexpr Real64 phi_rad = 0.79;
+        Real64 hr = epsilon * phi_rad * sigma * (std::pow(Tf, 2) + std::pow(Ta, 2)) * (Tf + Ta);
+        return 1.0 / (hc + hr);
+    }
+
+    // Function to calculate heat transfer resistance through air, when being naked, K m^2/W
+    Real64 Ra_un(Real64 Ts, Real64 Ta)
+    {
+        constexpr Real64 hc = 12.3;
+        constexpr Real64 phi_rad = 0.80;
+        Real64 hr = epsilon * phi_rad * sigma * (std::pow(Ts, 2) + std::pow(Ta, 2)) * (Ts + Ta);
+        return 1.0 / (hc + hr);
+    }
+
+    constexpr Real64 Za = 60.6 / 17.4;     // Pa m^2/W, mass transfer resistance through air, exposed part of skin
+    constexpr Real64 Za_bar = 60.6 / 11.6; // Pa m^2/W, mass transfer resistance through air, clothed part of skin
+    constexpr Real64 Za_un = 60.6 / 12.3;  // Pa m^2/W, mass transfer resistance through air, when being naked
+
+    constexpr Real64 tol = 1e-8;
+    constexpr Real64 maxIter = 100;
+    Real64 find_eqvar_phi(EnergyPlusData &state, Real64 Ta, Real64 RH)
+    {
+
+        Real64 phi = 0.84;
+        Real64 Pa = RH * pvstar(Ta);
+        Real64 Rs = 0.0387;
+        Real64 m = (Pc - Pa) / (Zs(Rs) + Za);
+        Real64 Ts;
+        int SolFla;
+        General::SolveRoot(
```

Comment:
Yes, this is the way.  If you want to make it slightly clearer, you _can_ define the lambda first so that you have plenty of room to name things nicely, and then just pass the lambda in as the argument.  It's fine as an argument here though.

Reply:
Make sense. Probably better as their own function in front